pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -3289,23 +3289,38 @@ private RetentionPolicies getRetentionPolicies(TopicName topicName, TopicPolicie
 
     }
 
-    protected MessageId internalTerminate(boolean authoritative) {
+    protected void internalTerminateAsync(AsyncResponse asyncResponse, boolean authoritative) {
```

Comment:
We'd better make this method return `CompletableFuture<MessageId>`, it's hard to get the response type for the rest API, only can be found here https://github.com/apache/pulsar/pull/14064/files#diff-66aeb65a64cbe7c541f013ae807c5bcbeab567bef77706c7ff2e0cbfe0d77eb1R3317